tests.texlive.fixedHashes: init

xworld21 opened the pull request:
## Description of changes
The assertion that all TeX Live packages have a fixed hash is time consuming and should only be checked when running tests. This should improve the evaluation speed of texlive by some small margin.

I apologise for the tiny conflict with another open texlive PR.

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

xworld21 committed:
tests.texlive.fixedHashes: init

The assertion that all TeX Live packages have a fixed hash is time
consuming and should only be checked when running tests.

veprbl commented:
@GrahamcOfBorg build tests.texlive

veprbl commented:
```
error: builder for '/nix/store/hpwypzd9jm8ls3ary9kmdngwyl02ysy0-texlive-test-license.drv' failed with exit code 1;
       last 10 log lines:
       >
       > license info for scheme-infraonly is incorrect! Note that order is enforced.
       > saved: [ gpl2 lgpl21 ]
       > correct: [ lgpl21 ]
       >
       >
       > license info for scheme-minimal is incorrect! Note that order is enforced.
       > saved: [ free gpl1Only gpl2 gpl2Plus knuth lgpl21 lppl1 lppl13c mit ofl publicDomain ]
       > correct: [ free gpl1Only gpl2Plus knuth lgpl21 lppl1 lppl13c mit ofl publicDomain ]
       >
```

xworld21 commented:
> license info for scheme-infraonly is incorrect! Note that order is enforced.
>        > saved: [ gpl2 lgpl21 ]
>        > correct: [ lgpl21 ]

This was spotted by @apfelkuchen6 and fixed in #247798 – where I was patching `texlive.infra`, so I added the `gpl2` license while I was there.

xworld21 commented:
@GrahamcOfBorg build tests.texlive.licenses tests.texlive.fixedHashes

veprbl commented:
@GrahamcOfBorg build tests.texlive

xworld21 commented:
Further motivation, for the record: the test must be moved also because the fixed hashes should be enforced on `texlive` as packaged in Nixpkgs. Overriding `texlive` outside of Nixpkgs should not require additional fixed hashes which is a side effect of the current `assert`.
